ranger/default.go

```diff
@@ -0,0 +1,390 @@
+	u, err := url.ParseRequestURI(envVarOrString(baseURLEnvVar, defaultBaseURL))
+	if err != nil {
+		err = fmt.Errorf("%w: failed parsing %s: %s", ErrBadConfig, baseURLEnvVar, err)
+		return func(rng *Ranger) (optFollowup, error) { return nil, err }
+	}
+
+	return func(rng *Ranger) (optFollowup, error) {
+		return func() error {
+			if rng.p == nil {
+				if _, err := DefaultParser(os.DirFS("."))(rng); err != nil {
+					return err
+				}
+			}
+
+			args := []resp.ResponderOptFn{
+				resp.WithRootUrl(u.String()),
+				resp.WithLogger(rng.Logger),
+				resp.WithParser(rng.p),
+				resp.WithAuthTemplate(defaultAuthedTmpl),
+				resp.WithUnauthTemplate(defaultUnauthedTmpl),
+				resp.WithSessionKey(rng.Keyring.SessionKey()),
+				resp.WithUserSessionKey(rng.Keyring.CurrentUserKey()),
+			}
+			for _, opt := range opts {
+				args = append(args, opt)
+			}
+
+			r := resp.NewResponder(args...)
+
+			fn, err := WithResponder(r)(rng)
+			if err != nil {
+				return err
+			}
+
+			return fn()
+		}, nil
+	}
+}
+
+// DefaultRouter constructs a RangerOption that returns a followup option
+// configuring the *Ranger.Router to be used by the web server.
+func DefaultRouter() RangerOption {
+	return func(rng *Ranger) (optFollowup, error) {
+		var fn func() error
+		var err error
+		fn, err = DefaultResponder()(rng)
+		if err != nil {
+			return nil, err
+		}
+
+		r := router.NewRouter(rng.Env.String())
+		r.OnEveryRequest(
+			middleware.RateLimit(middleware.NewVisitors()),
+			middleware.ForceHTTPS(rng.Env.String()),
+			middleware.RequestID(rng.Keyring.Key(defaultRequestIDCtxKey.Key())),
+			middleware.InjectIPAddress(),
+			middleware.LogRequest(rng),
+			middleware.InjectSession(rng.sess, rng.Keyring.SessionKey()),
+			middleware.CurrentUser(
+				rng.Responder,
+				rng,
+				rng.Keyring.SessionKey(),
+				rng.Keyring.CurrentUserKey(),
+			),
+		)
+
+		return func() error {
+			if err := fn(); err != nil {
+				return err
+			}
+
+			fn, err = WithRouter(r)(rng)
+			if err != nil {
+				return err
+			}
+
+			return fn()
+		}, nil
+	}
+}
+
+// DefaultSessionStore constructs a RangerOption that configures the SessionStore
+// to be used for storing session data.
+//
+// DefaultSessionStore relies on two env vars:
+// - "SESSION_AUTH_KEY"
+// - "SESSION_ENCRYPTION_KEY"
+//
+// These must be valid hex encoded values.
+//
+// If these values do not exist, DefaultSessionStore will generate new, random keys
+// and save those to the env var file for later use.
+func DefaultSessionStore(opts ...session.ServiceOpt) RangerOption {
+	return func(rng *Ranger) (optFollowup, error) {
+		args := []session.ServiceOpt{
+			session.WithCookie(),
+			session.WithMaxAge(3600 * 24 * 7),
+		}
+
+		for _, opt := range opts {
+			args = append(args, opt)
+		}
+
+		auth := os.Getenv(sessionAuthKeyEnvVar)
+		if auth == "" {
+			k := fmt.Sprintf("%x", securecookie.GenerateRandomKey(64))
+			if err := writeToEnvFile(sessionAuthKeyEnvVar, k); err != nil {
```

Comment:
I appreciate the ease of use here, but I think we ought to make this difficult for the user. I'd gander a public review would have all sorts of qualms with writing to the .env. We also want to support multiple keys for rotation purposes